src/main/kotlin/com/sourcegraph/common/ui/DumbAwareBGTAction.kt

```diff
@@ -0,0 +1,24 @@
+package com.sourcegraph.common.ui
+
+import com.intellij.openapi.actionSystem.ActionUpdateThread
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.util.NlsActions
+import javax.swing.Icon
+
+abstract class DumbAwareBGTAction : DumbAwareAction {
+
+  constructor() : super()
+
+  constructor(icon: Icon?) : super(icon)
+
+  constructor(text: @NlsActions.ActionText String?) : super(text)
+
+  constructor(
+      text: @NlsActions.ActionText String?,
+      description: @NlsActions.ActionDescription String?,
+      icon: Icon?
+  ) : super(text, description, icon)
+
+  @Suppress("MissingRecentApi")
```

Comment:
nice 🚀

Reply:
btw, can you provide a comment on this class? to explain why we need it, why we need to suppress this warning, and when can we drop it.

Reply:
Comments added.